### PR TITLE
Use proper logging mechanism instead of notify on macOS

### PR DESCRIPTION
**Pull request by hoechenberger**
The notifications often go unnoticed if a focus mode is active.

And if they don't go unnoticed, we found they're irritating to users, as clicking on them spawns a script editor.

Lastly, the emitted messages are not logged inside the installer.

To resolve these issues, the changes here replace all `notify` calls with proper logger invocations.

### CODE REVIEW

**Comment by hoechenberger** [comment]
I've signed the CLA. It appears it needs to be manually verified first?

**Comment by kenodegard** [comment]
@conda-bot check

**Comment by kenodegard** [comment]
@conda-bot check

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,4 +1,15 @@
 .. current developments
+
+XXXX-XX-XX   X.X.X:
+===================
+
+Enhancements:
+-------------
+
+* The graphical macOS installers now use the macOS logging system to log
```
Comment (hoechenberger) [inline CHANGELOG.rst:9]:
```suggestion
* macOS installers now use the macOS logging system to log
```

**Comment by jaimergp** [comment]
I added the notifications because we got feedback from users that thought the installer had hung. I get your point about intrusiveness and lack of logs, so I'd propose if we can make this an option? Something like:

```yaml
osx_pkg_progress_notifications: true  # default is false
```

The normal `echo` calls are logged too, by the way. I am not sure if using `logger` is advantageous in other ways, though.

**Comment by hoechenberger** [comment]
> The normal `echo` calls are logged too, by the way. I am not sure if using `logger` is advantageous in other ways, though.

Are you sure about that? IIRC only with those logger calls I could properly see all messages in the GUI installer log window. But I'd have to check again.

**Comment by hoechenberger** [comment]
> I added the notifications because we got feedback from users that thought the installer had hung. I get your point about intrusiveness and lack of logs, so I'd propose if we can make this an option? Something like:
> 
> ```yaml
> osx_pkg_progress_notifications: true  # default is false
> ```

I've received user feedback that the notifications were confusing.

It would be great if we could find a way to display "proper" notifications somehow and not have to abuse AppleScript for this stuff…

**Comment by jaimergp** [comment]
@hoechenberger Would it be ok if I push some changes to your branch introducing the configuration option to enable notifications (off by default)? This would be the ideal case for a plugin, but we don't have support for that (maybe in the future?), so one more config key would have to do now... If you prefer I open a 2nd PR as an alternative implementation, that's ok too!

**Comment by hoechenberger** [comment]
@jaimergp Please feel free to push changes here, I'm currently busy with other stuff and therefore didn't have the time to come back to this PR

**Review by jaimergp — DISMISSED** [review]
I'd like to see this as an opt-in setting (notifications off by default). I'll implement it next week, please don't merge yet :)

File: CONSTRUCT.md [hunk 2]
```diff
@@ -380,6 +380,16 @@ available and set to the destination causing the error. Defaults to:
...
 
+## `progress_notifications`
+
+_required:_ no<br/>
+_type:_ boolean<br/>
+Whether to show UI notifications on PKG installers. On large installations,
+the progress bar reaches ~90% very quickly and stays there for a long time.
+This might look like the installer froze. This option enables UI notifications
+so the user receives updates after each command executed by the installer.   
+(MacOS only)
```
Comment (hoechenberger) [inline CONSTRUCT.md:391]:
```suggestion
(macOS only) 
```

File: constructor/construct.py [hunk 3]
```diff
@@ -298,6 +298,14 @@
...
 
 (MacOS only)
+'''),
+
+    ('progress_notifications', False, bool, '''
+Whether to show UI notifications on PKG installers. On large installations,
+the progress bar reaches ~90% very quickly and stays there for a long time.
+This might look like the installer froze. This option enables UI notifications
+so the user receives updates after each command executed by the installer.   
+(MacOS only)
```
Comment (hoechenberger) [inline constructor/construct.py:308]:
```suggestion
(macOS only) 
```

**Comment by hoechenberger** [comment]
I think new commits should always refer to `macOS` (which is the correct spelling). We should do a separate PR to fix existing spelling…

File: CHANGELOG.rst [hunk 4]
```diff
@@ -6,6 +6,9 @@ XXXX-XX-XX   X.X.X:
 Enhancements:
 -------------
 
+* macOS installers now use the macOS logging system to log
+  messages. Previously, notification badges would be displayed. (#535)
```
Comment (jaimergp) [inline CHANGELOG.rst:10]:
```suggestion
* PKG installers now default to the macOS logging system to log messages.
  UI notifications can be enabled with `progress_notifications`
  (off by default). (#535)
```

**Comment by jaimergp** [comment]
Hm, we have a `news/` folder but also a `CHANGELOG.rst`. I think the CHANGELOG is only updated by the release automations though, so we should turn the changes in CHANGELOG into a news piece :D I'll go for it.

File: CHANGELOG.rst [hunk 5]
```diff
@@ -6,6 +6,10 @@ XXXX-XX-XX   X.X.X:
 Enhancements:
 -------------
 
+* PKG installers now default to the macOS logging system to log messages.
+  UI notifications can be enabled with `progress_notifications`
+  (off by default). (#535)
+
```
Comment (jaimergp) [inline CHANGELOG.rst:12]:
```suggestion
```

**Review by hoechenberger — COMMENTED** [review]
Very nice. I cannot approve since I was the one who originally opened this PR.

**Review by larsoner — APPROVED** [review]
Thanks @hoechenberger @jaimergp !

**Pull request merged**